xbmc/addons/AddonManager.cpp

```diff
@@ -848,14 +849,23 @@ bool CAddonMgr::CanAddonBeInstalled(const AddonPtr& addon)
 
 bool CAddonMgr::CanUninstall(const AddonPtr& addon)
 {
-  return addon && CanAddonBeDisabled(addon->ID()) &&
-      !StringUtils::StartsWith(addon->Path(), CSpecialProtocol::TranslatePath("special://xbmc/addons"));
+  return addon && CanAddonBeDisabled(addon->ID()) && !IsSystemAddon(addon->ID()) &&
+         !StringUtils::StartsWith(addon->Path(),
+                                  CSpecialProtocol::TranslatePath("special://xbmc/addons"));
 }
 
 bool CAddonMgr::IsSystemAddon(const std::string& id)
 {
   CSingleLock lock(m_critSection);
-  return std::find(m_systemAddons.begin(), m_systemAddons.end(), id) != m_systemAddons.end();
+  return IsOptionalSystemAddon(id) ||
+         std::find(m_systemAddons.begin(), m_systemAddons.end(), id) != m_systemAddons.end();
+}
```

Comment:
usually i would not change behavior of an existing function unless it is acting faulty. e.g. if you have a codepath where `IsSystemAddon` must not return optional addons for some reason. but it's feasible here i think, as we have only 3 calls.

Reply:
Why would an optional add-on ever be in `m_systemAddons`? Are you sure changing this function is required?

Reply:
Yes. Optional system addons are still system addons (both have system_origin as origin) should be treated the same for nearly evey case except they can be disabled